privacyidea/api/realm.py

```diff
@@ -375,10 +431,101 @@ def delete_realm_api(realm=None):
           },
           "version": "privacyIDEA unknown"
         }
-
     """
     ret = delete_realm(realm)
-    g.audit_object.log({"success": ret,
+    g.audit_object.log({"success": ret > 0,
                         "info": realm})
 
     return send_result(ret)
+
+
+# TODO: Flask supports conversion of the nodeid to a UUID, but in get_all_params()
+#  we put everything (including the viewargs) in the param dict and try to "unquote"
+#  all entries. This fails for the UUID type. We should probably remove the viewargs from
+#  the param dict and handle them separately in the corresponding API function.
+@realm_blueprint.route('/<string:realm>/node/<string:nodeid>', methods=['POST'])
+@log_with(log)
+@prepolicy(check_base_action, request, ACTION.RESOLVERWRITE)
+def set_realm_node_api(realm, nodeid):
+    """
+    This call creates or reconfigures a realm with node specific settings.
+
+    The realm contains a list of resolvers and corresponding priorities per node.
+    If the node UUID can not be found in the database, the realm will be
+    configured without a specific node.
+    In the result it returns a list of added resolvers and a list of
+    resolvers, that could not be added.
+
+    :arg string realm: The unique name of the realm
+    :arg string nodeid: The UUID of the node
+    :<json list resolvers: A JSON list consisting of objects with the resolver name
+      and priority
+    :reqheader PI-Authorization: The authorization token
+
+    **Example request**:
+
+    Create a new realm ``newrealm``, that consists of the resolvers
+    ``resolver1`` and ``resolver2`` on the node with
+    UUID 8e4272a9-9037-40df-8aa3-976e4a04b5a9:
+
+    .. sourcecode:: http
+
+       POST /realm/newrealm/node/8e4272a9-9037-40df-8aa3-976e4a04b5a9 HTTP/1.1
+       Host: example.com
+       Accept: application/json
+       Content-Type: application/json
+
+       [
+         {
+           "name": "resolver1",
+           "priority": 1
+         },
+         {
+           "name": "resolver2",
+           "priority": 2
+         }
+       ]
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+       {
+          "id": 1,
+          "jsonrpc": "2.0",
+          "result": {
+                    "status": true,
+                    "value": {
+                        "added": ["resolver1", "resolver2"],
+                        "failed": []
+                    }
+          }
+          "version": "privacyIDEA unknown"
+       }
+
+    .. versionadded:: 3.10 Node specific realm configuration
+    """
+    if not check_node_uuid_exists(nodeid):
+        log.warning(f"Node with UUID {nodeid} does not exist in the database!")
+        raise ParameterError(_("The given node does not exist!"))
+
+    data = request.json
+    resolvers = []
+    for res in data:
```

Comment:
The list of resolvers is not handled correctly in `before_request()`. It assumes the resolvers are json, but since it is a list it sets {"name": "priority"}` to `request.all_data`.  Additionally, it throws a `TypeError` if only a single resolver without a priority is passed, because it fails to handle a list with a single entry somehow as json.

Reply:
Done in 200d355

Reply:
Done in 658afd5